Yoast/yoast-acf-analysis
# CS: start using YoastCS 0.5 & PHPCS 3.3.0

## Summary

This PR can be summarized in the following changelog entry:
* _N/A_

## Relevant technical choices:

This PR moves this repo onto [YoastCS](https://github.com/Yoast/yoastcs) 0.5 in combination with [PHPCS 3.3.0](https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.3.0).

Please see the individual commits for details and complete documentation about the changes.

* Changed the ruleset base from WPCS to YoastCS and fixed the YoastCS dependency to the `0.5.*` range.
* Renamed and updated the ruleset file.
    * Added configuration for various WPCS sniffs to increase their usefulness.
        See [WPCS Customizable sniff properties](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties) for detailed information.
    * Added a few temporary exclusions:
        - One for a WPCS bug which will be fixed in the WPCS `1.0.0` release;
        - And a connected set of exclusions for documentation issues which should be fixed. See #152.
            Most of these issues have been excluded from being reported, though for one, the `error` has been downgraded to a `warning`.
* Split the Travis run into two: plugin files and test files.
    The plugin files in the ACF analysis plugin have to be compatible with PHP 5.2, however, the test files have been set up to work only on PHP 5.6+ so reports about incompatible PHP features used would be reported without this split.
    Whether it's a good idea not to test the plugin on older PHP versions is another matter, but not one which should be addressed in this PR.

## Test instructions

This PR can be tested by following these steps:
* Run `composer check-cs` without arguments to see that the library code is clean (other than the above mentioned warnings).
